### PR TITLE
Fix the error when save adv model and use modelcheckpoint callback

### PR DESCRIPTION
**Pull request by wangbingnan136**

### CODE REVIEW

**Comment by wangbingnan136** [comment]
> @wangbingnan136[，你能看一下并在上面的评论](https://github.com/tensorflow/neural-structured-learning/pull/114#issuecomment-1129201056)中签署贡献者许可协议吗？

image
It looks like I have already signed

**Comment by wangbingnan136** [comment]
I think the checks were done,sorry this is my first time make a contribution to tensorflow~

File: neural_structured_learning/keras/adversarial_regularization.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 import collections
 import functools
 import types
+from keras.utils import traceback_utils
```
Comment (csferng) [inline neural_structured_learning/keras/adversarial_regularization.py:23]: Please move these two import lines to line 27 (after "import attr") and swap their order. The idea is that the imports from the standard library goes before the imports from third-party libraries, and the imports in the same group are ordered alphabetically.

File: neural_structured_learning/keras/adversarial_regularization.py [hunk 2]
```diff
@@ -20,6 +20,8 @@
...
+from keras.saving import save
```
Comment (csferng) [inline neural_structured_learning/keras/adversarial_regularization.py:24]: What is the earliest version of Keras providing these two modules (`save` and `traceback_utils`)? Please add a line to the [requirements file](https://github.com/tensorflow/neural-structured-learning/blob/master/neural_structured_learning/requirements.txt).

**Review by csferng — CHANGES_REQUESTED** [review]

File: neural_structured_learning/examples/notebooks/Adversarial_regularization_for_image_classification save and load the adv model.ipynb [hunk 3]
```diff
@@ -0,0 +1,710 @@
+{
```
Comment (csferng) [inline neural_structured_learning/examples/notebooks/Adversarial_regularization_for_image_classification save and load the adv model.ipynb:1]: <p>Thanks for adding an example notebook. I think it is great to demonstrate the saving and loading functionalities. Please update the title and the overview section to focus on that. Also maybe please rename this file to match the new title.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/neural-structured-learning/pull/114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (csferng) [inline neural_structured_learning/examples/notebooks/Adversarial_regularization_for_image_classification save and load the adv model.ipynb:1]: <p>How about adding some evaluation to demonstrate that the loaded model performs the same as the model before saving?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/neural-structured-learning/pull/114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

**Comment by wangbingnan136** [comment]
ok,I got it!

**Comment by wangbingnan136** [comment]
It is done

**Comment by csferng** [comment]
Thanks for the fix! Let me polish the example notebook a bit before merging the PR.

**Comment by wangbingnan136** [comment]
Okk

**Comment by wangbingnan136** [comment]
I found that when we load the model，the "fit" and "predict" methods works as expected.But the "evalaute" has some bugs.I will fix this problem when I am not busy.

**Comment by csferng** [comment]
Sorry for the delay. I made some updates on the notebook and will proceed to merge this PR.

Regarding the bug on `evaluate`, what I encountered is that the accuracy number was off for the loaded model (`0.09` vs. `0.98`). The root cause turns out to be that `metrics=['acc']` is interpreted differently. (Keras has some internal heuristic to distinguish what "accuracy" means to a model.) What we expect is `tf.keras.metrics.SparseCategoricalAccuracy`, which is the case for the `adv_model`. But the loaded model somehow got `tf.keras.metrics.CategoricalAccuracy`, so the evaluation result became different.  To workaround this issue, I changed `metrics=['acc']` to `metrics=[tf.keras.metrics.SparseCategoricalAccuracy()]`, which then produced consistent result before and after saving/loading.
